### PR TITLE
Always set parent of agent class loader to bootstrap class loader

### PR DESCRIPTION
**Pull request by laurit**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/5156

### CODE REVIEW

File: javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentInitializer.java [hunk 1]
```diff
@@ -100,31 +98,8 @@ public static ClassLoader getExtensionsClassLoader() {
...
-     compatible with java 8.
-    */
-    Method method = ClassLoader.class.getDeclaredMethod("getPlatformClassLoader");
-    return (ClassLoader) method.invoke(null);
-  }
-
-  public static boolean isJavaBefore9() {
-    return System.getProperty("java.version").startsWith("1.");
+  private static ClassLoader createAgentClassLoader(String innerJarFilename, File javaagentFile) {
+    return new AgentClassLoader(javaagentFile, innerJarFilename, null);
```
Comment (trask) [inline javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentInitializer.java:102]: any reason to keep the `parent` argument in the constructor?

**Review by trask — APPROVED** [review]
👍

File: javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentInitializer.java [hunk 2]
```diff
@@ -100,31 +98,8 @@ public static ClassLoader getExtensionsClassLoader() {
...
    * @return Agent Classloader
    */
-  private static ClassLoader createAgentClassLoader(String innerJarFilename, File javaagentFile)
-      throws Exception {
-    ClassLoader agentParent;
-    if (isJavaBefore9()) {
-      agentParent = null; // bootstrap
-    } else {
-      // platform classloader is parent of system in java 9+
-      agentParent = getPlatformClassLoader();
```
Comment (anuraaga) [inline javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentInitializer.java:110]: Any idea why we were doing this before?

**Review by anuraaga — APPROVED** [review]

File: javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentInitializer.java (hunk 2 above)
Comment (trask) [reply javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentInitializer.java:110]: this was inherited, my best guess is for access to `java.sql` package for some reason(?)

**Pull request merged**